code/game/gamemodes/miniantags/pulsedemon/pulsedemon.dm

```diff
@@ -0,0 +1,821 @@
+	SIGNAL_HANDLER
+	if(cable_images[T])
+		var/list/turf_images = cable_images[T]
+		for(var/image/current_image in turf_images)
+			client?.images -= current_image
+		turf_images.Cut()
+	else
+		cable_images[T] = list()
+
+	for(var/obj/structure/cable/C in T)
+		var/image/cable_image = image(C, C, layer = ABOVE_LIGHTING_LAYER, dir = C.dir)
+		cable_image.plane = ABOVE_LIGHTING_PLANE
+		cable_images[T] += cable_image
+		client?.images += cable_image
+
+/mob/living/simple_animal/demon/pulse_demon/proc/apc_deleted_handler(obj/machinery/power/apc/A, force)
+	SIGNAL_HANDLER
+	hijacked_apcs -= A
+
+/mob/living/simple_animal/demon/pulse_demon/Destroy()
+	cable_images.Cut()
+	apc_images.Cut()
+
+	controlling_area = null
+	current_bot = null
+	current_cable = null
+	current_power = null
+	current_robot = null
+	current_weapon = null
+	apc_being_hijacked = null
+	hijacked_apcs = null
+	hijacked_robots = null
+
+	return ..()
+
+/mob/living/simple_animal/demon/pulse_demon/Login()
+	. = ..()
+	update_cableview()
+
+/mob/living/simple_animal/demon/pulse_demon/proc/make_pulse_antagonist(demon)
+	SIGNAL_HANDLER
+	mind.assigned_role = SPECIAL_ROLE_DEMON
+	mind.special_role = SPECIAL_ROLE_DEMON
+	give_objectives()
+
+/mob/living/simple_animal/demon/pulse_demon/vv_edit_var(var_name, var_value)
+	switch(var_name)
+		if("glow_color")
+			update_glow()
+		if("charge")
+			// automatically adjusts maxcharge to allow the new value
+			adjust_charge(var_value - charge, TRUE)
+			return TRUE
+	return ..()
+
+/mob/living/simple_animal/demon/pulse_demon/forceMove(atom/destination)
+	var/old = loc
+	. = ..()
+	current_weapon = null
+	current_robot = null
+	if(current_bot)
+		current_bot.hijacked = FALSE
+	current_bot = null
+	if(istype(old, /obj/item/stock_parts/cell))
+		var/obj/item/stock_parts/cell/C = old
+		// only set rigged if there are no remaining demons in the cell
+		C.rigged = !(locate(/mob/living/simple_animal/demon/pulse_demon) in old)
+	if(istype(loc, /obj/item/stock_parts/cell))
+		var/obj/item/stock_parts/cell/C = loc
+		C.rigged = FALSE
+
+/mob/living/simple_animal/demon/pulse_demon/proc/give_objectives()
+	if(!mind)
+		return
+	mind.wipe_memory()
+	to_chat(src, "<br>")
+	to_chat(src, "<span class='warning'><font size=3><b>You are a pulse demon.</b></font></span>")
+	to_chat(src, "<b>A being made of pure electrical energy, you travel through the station's wires and infest machinery.</b>")
+	to_chat(src, "<b>Navigate the station's power cables to find power sources to steal from, and hijack APCs to interact with their connected machines.</b>")
+	to_chat(src, "<b>If the wire or power source you're connected to runs out of power you'll start losing health and eventually die, but you are otherwise immune to damage.</b>")
+	to_chat(src, "<span class='motd'>For more information, check the wiki page: ([GLOB.configuration.url.wiki_url]/index.php/Pulse_Demon)</span>")
+
+	var/i = 1
```

Comment:
Could be renamed to "obective_number" or something like that to make it more clean to read

Reply:
done